c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -1,22 +1,606 @@
+    aliases.insert({COMPLEMENTOFDEFAULT, complementOfDefault});
+
+    // The content of the default list is determined at compile time and normally corresponds to
+    // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+    aliases.insert({DEFAULT, this->parse("ALL:!COMPLEMENTOFDEFAULT:!eNULL")});
+  }
+
+  std::vector<Cipher> parse(std::string_view expression) const {
+    /**
+     * All ciphers by their openssl alias name.
+     */
+    auto elements = this->splitCipherSuiteString(expression);
+    std::vector<Cipher> ciphers;
+    std::vector<Cipher> removedCiphers;
+    for (auto &element : elements) {
+
+      if (element.rfind(DELETE, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          this->remove(aliases, ciphers, alias);
+        }
+      } else if (element.rfind(EXCLUDE, 0) == 0) {
+        auto alias = element.substr(1);
+        auto found = aliases.find(alias);
+        if (found != aliases.end()) {
+
+          auto toAdd = found.operator->()->second;
+          removedCiphers.insert(removedCiphers.end(), toAdd.begin(), toAdd.end());
+        }
+      } else if (element.rfind(TO_END, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          this->moveToEnd(aliases, ciphers, alias);
+        }
+      } else if ("@STRENGTH" == element) {
+        this->strengthSort(ciphers);
+        break;
+      } else if (aliases.find(element) != aliases.end()) {
+        this->add(aliases, ciphers, element);
+      } else if (element.find(AND) != std::string::npos) {
+        auto intersections = this->split(element, "+\\");
+        if (intersections.size() > 0) {
+          auto found = aliases.find(intersections[0]);
+          if (found != aliases.end()) {
+            auto result{found.operator->()->second};
+            for (int i = 1; i < intersections.size(); i++) {
+              auto alias = aliases.find(intersections[i]);
+              if (alias != aliases.end()) {
+                // make `result` only contain the aliases from `alias`
+                result.erase(std::remove_if(result.begin(), result.end(),
+                                            [&](auto x) {
+                                              return std::find(alias->second.begin(),
+                                                               alias->second.end(),
+                                                               x) != alias->second.end();
+                                            }),
+                             result.end());
+              }
+            }
+            // Add all of `result` onto `ciphers`
+            ciphers.insert(ciphers.end(), result.begin(), result.end());
+          }
+        }
+      }
+    }
+    // Remove all ciphers from `ciphers` which are contained in `removedCiphers`
+    ciphers.erase(std::remove_if(ciphers.begin(), ciphers.end(),
+                                 [&removedCiphers](Cipher c) {
```

Comment:
Avoid copying the `c` argument by turning it into a reference.
```suggestion
                                 [&removedCiphers](auto &c) {
```